### PR TITLE
Khalil/2706 update middleware errors

### PR DESCRIPTION
**Pull request by kc1116**
This PR improves error documentation for the middleware struct.

### CODE REVIEW

**Review by jordanschalm — APPROVED** [review]
Looks good, thanks for the improvements 🎸

File: network/p2p/middleware.go [hunk 1]
```diff
@@ -340,6 +351,17 @@ func (m *Middleware) stop() {
...
 // a more efficient candidate.
+//
+// The following benign errors can be returned from libp2p:
+// - the peer ID for the target node ID cannot be found.
+// - the msg size exceeds result returned from unicastMaxMsgSize(msg)
+// - the libP2P node fails to publish the message.
+// - the libP2P node fails to create the stream.
+// - setting write deadline on the stream fails.
+// - the gogo protobuf writer fails to write the message.
+// - flushing the stream fails.
```
Comment (peterargue) [inline network/p2p/middleware.go:362]: I think this list is overly specific and leaks too much information about the implementation. You could simplify it to something like this without losing too much context.

```suggestion
// The following benign errors can be returned:
// - he peer ID for the target node ID cannot be found.
// - the msg size was too large.
// - failed to send message to peer.
```

**Review by peterargue — APPROVED** [review]
Thanks for adding these @kc1116. Added one comment, but otherwise looks good

File: network/p2p/middleware.go (hunk 1 above)
Comment (kc1116) [reply network/p2p/middleware.go]: https://github.com/onflow/flow-go/pull/2726/commits/3dac90936ad08f9dd47396507383ab559641c904

**Comment by kc1116** [comment]
bors merge

**Pull request merged**